docs/modules/usage/how-to/custom-sandbox-guide.md

```diff
@@ -12,53 +12,131 @@ You have two options for customization:
 1. Use an existing image with the required software.
 2. Create your own custom Docker image.
 
-If you choose the first option, you can skip the `Create Your Docker Image` section.
+## Important Requirements
 
-## Create Your Docker Image
+Before proceeding with either option, note these key requirements:
+
+1. The base image must be Debian-based
+2. The user in the container MUST be `root` for proper functionality
+3. Any installed packages should be available system-wide, not just for specific users
 
-To create a custom Docker image, it must be Debian based.
+## Create Your Docker Image
 
-For example, if you want OpenHands to have `ruby` installed, create a `Dockerfile` with the following content:
+Here's a complete example of a custom runtime Dockerfile:
 
 ```dockerfile
 FROM debian:latest
 
 # Install required packages
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y \
+    ruby \
+    python3 \
+    nodejs \
+    && rm -rf /var/lib/apt/lists/*
+
+# Ensure we're running as root
+USER root
 ```
 
-Save this file in a folder. Then, build your Docker image (e.g., named custom-image) by navigating to the folder in
-the terminal and running::
+Save this file as `docker/runtime.Dockerfile`. Then, build your Docker image using one of these methods:
```

Comment:
we should probably note that a nested docker/ dir is not required.

Reply:
Not sure if we want to take into account people that might not be too familiar with docker and dockerfiles.
In the previous version, it said to navigate to the folder in the terminal and then run the command. In the new version, it doesn't specify where to run the command from.

Reply:
@mamoodi sounds like you want a flattened example. I'm fine with that. Let's update the documentation so that we don't have any nested directory structures.